jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -108,18 +109,41 @@ public static boolean getDefaultUseCaches()
      */
     public static Resource.Mount mountIfNeeded(URI uri)
     {
-        if (uri == null || uri.getScheme() == null)
+        if (uri == null)
+            return null;
+        String scheme = uri.getScheme();
+        if (scheme == null)
+            return null;
+        if (!isJar(uri))
             return null;
         try
         {
-            return (uri.getScheme().equalsIgnoreCase("jar")) ? FileSystemPool.INSTANCE.mount(uri) : null;
+            if (scheme.equalsIgnoreCase("jar") || scheme.equalsIgnoreCase("file"))
```

Comment:
why are you mounting file URIs?  Is it because somethings it has not been converted to a jar:file URI yet? In which case I'd prefer we always did that.

Reply:
If you want to mount the jar pointed to by a `file:` URI, I'd prefer to have a `mountJarIfNeeded(URI jarFileUri)` helper method that explicitly does that.